### PR TITLE
Enable `--experimental_allow_unresolved_symlinks` by default.

### LINKED ISSUE #10298: Stabilize unresolved symbolic links (--experimental_allow_unresolved_symlinks)
### Description of the problem / feature request:

https://github.com/bazelbuild/bazel/issues/9075 added support for unresolved symlinks, via the `actions.declare_symlink()` and `actions.symlink()` functions. That functionality is currently gated by `--experimental_allow_unresolved_symlinks`.

I've discovered that unresolved symlinks are critical for supporting hermetic Node.JS builds. Specifically, Node's use of symlinks for module resolution semantics mean that we need to be able to create symlink artifacts that reference dependencies and can be used as runfiles, but exist outside the runfiles root.

What are the current blockers to stabilizing unresolved symlinks? Is there a list of known remaining work to get them enabled by default?

cc @lberki @laszlocsomor

### PR DESCRIPTION
**Pull request by meisterT**
Fixes #10298.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java [hunk 1]
```diff
@@ -650,7 +650,7 @@ public OutputDirectoryNamingSchemeConverter() {
 
   @Option(
       name = "experimental_allow_unresolved_symlinks",
```
Comment (fmeum) [inline src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:652]: How about making this the old name and using just `allow_unresolved_symlinks` as the new name? Having to disable a default option via a flag called `experimental` could be confusing for users.

Comment (meisterT) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:652]: I am not sure the flag will stick around or whether people should actively toggle it; if it doesn't stick around then we should keep the `experimental_` prefix. 

 And I am not the person to make that call - I am just doing the bare minimum to enable this in Bazel 6.0.

**Comment by meteorcloudy** [comment]
Sorry for the breakage! Should be fixed in https://github.com/bazelbuild/bazel/commit/e7b258c3d6f91a6b65136614c4f0360a35727fe2, please rebase and retry.

**Review by lberki — APPROVED** [review]

File: src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java (hunk 1 above)
Comment (lberki) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:652]: Realistically, I don't think we will flip this flag at Google anytime soon so I guess it'll have to stay?

Comment (meisterT) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:652]: For me the question is more whether we want people (both internally and externally) to change their default value. If not, let's keep the prefix?

Comment (lberki) [reply src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java:652]: I was scratching my head as to how to decide that. Your line of reasoning is quite nice, so I guess let's keep it?

**Pull request closed without merging**